data/scripts.js

```diff
@@ -103,7 +103,11 @@ exports.BattleScripts = {
 	 */
 	useMove: function (move, pokemon, target, sourceEffect, zMove) {
 		if (!sourceEffect && this.effect.id) sourceEffect = this.effect;
-		if (zMove || (sourceEffect && sourceEffect.isZ)) {
+		if (zMove && move.id === 'weatherball') {
+			this.singleEvent('ModifyMove', move, null, pokemon, target, move, move);
+			move = this.getZMoveCopy(move, pokemon);
+			if (move.type !== 'Normal') this.add('-message', `Breakneck Blitz turned into ${move.name} due to the weather!`);
```

Comment:
Yeah, that looks as if it should work.

FYI in the end I couldn't actually find anywhere else particularly suitable to put this code. (The next best place I could find was in `resolvePriority` of all places...)